pool/src/contract.rs

```diff
@@ -57,15 +59,35 @@ pub trait Pool {
     /// If the caller is not the admin
     fn update_pool(e: Env, backstop_take_rate: u64);
 
-    /// (Admin only) Initialize a reserve in the pool
+    /// (Admin only) Queues the initialization of a reserve in the pool
     ///
     /// ### Arguments
     /// * `asset` - The underlying asset to add as a reserve
     /// * `config` - The ReserveConfig for the reserve
     ///
     /// ### Panics
-    /// If the caller is not the admin or the reserve is already setup
-    fn init_reserve(e: Env, asset: Address, metadata: ReserveConfig) -> u32;
+    /// If the caller is not the admin
+    fn queue_init_reserve(e: Env, asset: Address, metadata: ReserveConfig);
+
+    /// (Admin only) Cancels the queued initialization of a reserve in the pool
+    ///
+    /// ### Arguments
+    /// * `asset` - The underlying asset to add as a reserve
+    ///
+    /// ### Panics
+    /// If the caller is not the admin or the reserve is not queued for initialization
+    fn cancel_init_reserve(e: Env, asset: Address);
```

Comment:
is this necessary?

Reply:
I think it's potentially problematic not to have one - consider someone sneaks through a bad DAO proposal or tricks a multisig into adding a reserve - note that executing the queued initialization is not permissioned